assets/js/hocs/with-searched-products.tsx

```diff
@@ -48,7 +48,7 @@ const withSearchedProducts = (
 					setIsLoading( false );
 				} )
 				.catch( setErrorState );
-		}, [ selected ] );
+		}, [] );
```

Comment:
This is an anti pattern that we fought a bit to get rid off and refactor away from, because it causes hard to debug bugs.
If you feel like `selected` is not going to change much here, I'd suggest you have it a ref then?

```js
const selectedRef = useRef( selected );

selectedRef.current;
```

This should achieve the same goal, and would not cause rerenders.

Reply:
Thanks for pointing that out @senadir, I'd be curious to know more about the types of bugs this causes! I've changed the implementation to use a ref as you suggested.

Reply:
Hey! This causes out of sync bugs, things running when not needed, and generally hard to find bugs, examples:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3314
https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3285